internal/artifact/uploader.go

```diff
@@ -1,10 +1,10 @@
-package agent
+package artifact
 
 import (
 	"github.com/buildkite/agent/v3/api"
 )
 
-type Uploader interface {
+type UploadDoer interface {
```

Comment:
taking any and all suggestions on a better name for this

surely this can't be it

Reply:
(there's a namespace collision with `artifact.Uploader`, which is the struct equivalent to `artifact.Downloader`, `.Searcher` etc)

Reply:
I think it's more conventional to reserve the agent noun (ie the ones that end in "er") for interfaces and rename the structs.

They could be something like `artifact.DownloadRequest` and `artifact.UploadRequest` as I think they correspond to a request to the API.

I guess that means you should rename about half the structs in this package, but that's what refactors like this are for!

Reply:
Or you could just combine the method names and agent nounifiy it: `URLUploader`, or `UploadURLer`. That's the approach `io.ReadWriter` takes.